### PR TITLE
Make remote commands smarter

### PR DESCRIPTION
**Pull request by mwleeds**
See the commit messages for the details. This is a fix for #787

### CODE REVIEW

File: app/flatpak-builtins-list-remotes.c [hunk 1]
```diff
@@ -194,10 +150,10 @@ gboolean
 flatpak_complete_list_remotes (FlatpakCompletion *completion)
 {
   g_autoptr(GOptionContext) context = NULL;
-  g_autoptr(FlatpakDir) dir = NULL;
 
   context = g_option_context_new ("");
-  if (!flatpak_option_context_parse (context, options, &completion->argc, &completion->argv, 0, &dir, NULL, NULL))
+  if (!flatpak_option_context_parse (context, options, &completion->argc, &completion->argv,
+                                     FLATPAK_BUILTIN_FLAG_ONE_DIR, NULL, NULL, NULL))
```
Comment (alexlarsson) [inline app/flatpak-builtins-list-remotes.c]: Flags should be 0 here, no?

File: app/flatpak-builtins-utils.c [hunk 2]
```diff
@@ -326,3 +326,59 @@ flatpak_load_gpg_keys (char **gpg_import,
...
+          g_autofree char *dir_name = flatpak_dir_get_name (dir);
+          g_print("%d) %s\n", i + 1, dir_name);
+        }
+      chosen = flatpak_number_prompt (1, dirs_with_remote->len, _("Which do you want to use?"));
+    }
+
+  if (out_dir)
+    {
+      if (dirs_with_remote->len == 0)
+        *out_dir = NULL;
```
Comment (alexlarsson) [inline app/flatpak-builtins-utils.c]: Every single user of this handles it by picking a random dir and letting some other operation fail. I think it makes a lot more sense to have this report an error here.

File: app/flatpak-builtins-ls-remote.c [hunk 3]
```diff
@@ -50,12 +51,33 @@ static GOptionEntry options[] = {
   { NULL }
 };
 
+typedef struct RemoteDirPair {
+  gchar *remote_name;
+  gchar *dir_id;
```
Comment (alexlarsson) [inline app/flatpak-builtins-ls-remote.c]: I don't think using the dir_id for this works. For instance, the user directory has a NULL directory id, but when you later call flatpak_dir_get_system_by_id() with this you'll get back the system dir. Why not have a ref to the actual directory

File: app/flatpak-builtins-list-remotes.c (hunk 1 above)
Comment (mwleeds) [reply app/flatpak-builtins-list-remotes.c]: yeah, not sure what I was thinking

Comment (mwleeds) [reply app/flatpak-builtins-list-remotes.c]: Now that I'm looking at it again I remember; it was that I was trying to maintain that the first commit didn't introduce any functional changes. So I'll move that change into a separate commit.

**Comment by mwleeds** [comment]
Alright @alexlarsson I addressed your comments, and I changed `flatpak_resolve_duplicate_remotes` so it doesn't segfault if a choice has to be made and it's run from a script.

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Comment by alexlarsson** [comment]
bot, retest this please

**Comment by alexlarsson** [comment]
```
# error: Remote "test-repo" not found
+ echo 'File '\''remote-modify-error-log'\'' doesn'\''t match regexp '\''No remote test-repo'\'''
```
I think you changed some string...

**Comment by mwleeds** [comment]
bot, retest this please

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**